File: python/13_Roman_to_Integer.py
#!/usr/bin/python
import sys
import string
import os

class Solution:
	def romanToInt(self, s):
		values = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1]
		numerals = ["M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"]
		out = 0
		for i in range(13):
			while s.startswith(numerals[i]):
			    # use replace for a string and pop or remove for a list
				s = string.replace(s, numerals[i], '', 1)
				out += values[i]
		return out



def main():
    s = Solution()
    s.romanToInt("DCXXI")

if __name__ == "__main__":
    main()

File: c++/13_Roman_to_Integer.cpp
#include <iostream>
#include <vector>
#include <map>
#include <cmath>
#include <string>

using namespace std;

class Solution {
public:
    int romanToInt(string s) {
    	int values[] = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
		string numerals[] = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
        int i = 0;
		int out = 0;
        int start = 0;
        while (i < 13 && start < s.size()){
            string tmp = s.substr(start, numerals[i].size());
            if (tmp == numerals[i]) {
                start += numerals[i].size();
                out += values[i];
            }
            else {
                i += 1;
            }
        }
		return out;
    }
};


int main() {
    Solution s;
    cout << s.romanToInt("MXIIIIII") << endl;


}
